src/diffusers/models/embeddings.py

```diff
@@ -545,11 +545,14 @@ def get_1d_rotary_pos_embed(
     assert dim % 2 == 0
 
     if isinstance(pos, int):
-        pos = np.arange(pos)
+        pos = torch.arange(pos)
+    if isinstance(pos, np.ndarray):
+        pos = torch.from_numpy(pos)  # type: ignore  # [S]
+
     theta = theta * ntk_factor
     freqs = 1.0 / (theta ** (torch.arange(0, dim, 2, dtype=freqs_dtype)[: (dim // 2)] / dim)) / linear_factor  # [D/2]
-    t = torch.from_numpy(pos).to(freqs.device)  # type: ignore  # [S]
-    freqs = torch.outer(t, freqs)  # type: ignore   # [S, D/2]
+    freqs = freqs.to(pos.device)
```

Comment:
I'd expect this to cause a sync as well since by default `arange` allocates on the CPU. One way to mitigate could be to 
a) use `pin_memory()` on `freqs` ahead of time and set `non_blocking=True`
b) do `arange` on the GPU right away (i.e. `torch.arange([...], device=pos.device)`).

Reply:
ohhh let's do `torch.arange([...], device=pos.device)`